docs/component_conventions.md

```diff
@@ -127,6 +127,32 @@ examples:
         <span>Some text</span>
 ```
 
+#### Yaml configuration for components that need contextual HTML
+
+If a component is only visible, or behaves differently, in a certain context
+the examples for it can be embedded within HTML using the embed option.
+eg.
+
+```ruby
+<button class="trigger-for-component">Click me</button>
+<%= render "my-hidden-by-default-component", { param: value } %>
+```
+
+To configure a HTML embed in the component yaml file you can specify `embed` at
+the root or individual examples:
+
+```yaml
+embed: |
+  <button class="trigger-for-component">Click me<button>
+  <%= component %>
+examples:
+  default:
```

Comment:
Really minor, but should we lose the `default:`?

Reply:
We need it here if we want to illustrate both the default one and a specific one. I know it looks a bit funky 😞

Reply:
Ah. Could we just put something small in here, then?

Reply:
I don't think there's anything we can really put since it doesn't need anything. It's just an aspect (or failing) of the config file design.

Reply:
No worries - it was only minor.